packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -484,6 +485,8 @@ export function createRootHooks<
   ): UseTRPCSuspenseInfiniteQueryResult<unknown, TError, unknown> {
     const hookResult = useInfiniteQuery(path, input, {
       ...opts,
+      // FIXME: How should we handle suspense hooks?
+      // @ts-expect-error - whatever?
```

Comment:
tests are passing - are we safe to use this option even if it's not public?

or should we use the built-in suspense hooks? how do we do with the prefetching logic we have in "our normal" hooks?

Reply:
cc @KATT @TkDodo

Reply:
Let's go for it I think -- WDYT @TkDodo?

Reply:
I would encourage you to please use the built-in `useSuspenseQuery`, `useSuspenseInfiniteQuery` and `useSuspenseQueries` hook. It's an implementation detail that they are now implemented with `suspense:true`, because I didn't get around to changing that, but it very well might change in the future in a minor version.

Reply:
Done!